enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md

```diff
@@ -0,0 +1,294 @@
+
+### Non-Goals
+
+1. This proposal assumes the sensitive content encapsulated in the K8S Secret is valid and correct for the user's purposes.
+No validation of the content for specific usage within a Pod will be done.  That said, this proposal does provide 
+for watching for updates to the K8S Secrets being stored on the host/node by the associated CSI Driver of this proposal.
+So if the cluster administrator realized on their own they have provided bad data, and then corrects it, this proposal
+will watch for, discover, and propagate the update to the host/node filesystem.
+
+2. Protecting this content from being viewed/copied by users. If it's available for the Pod to use, the user who can 
+create a Pod can also read / view / exfiltrate the Secret contents. It is technically impossible to avoid that.
+
+3. 4.x customer feedback has also revealed a desire for namespace level granularity when injecting/consuming
+K8S Secret information available on the underlying host/node.  While this proposal will articulate some thought on 
+how to do this, ultimately a solution based on a given Secret utilized across multiple namespaces is deemed "meets
+minimum" for this proposal, and per namespace secret injection is out of scope.   
+
+## Proposal
+
+### User Stories [optional]
+
+1. As a cluster admin with users who needs to have their Pod access a finite set of content on a host/node in read only 
+fashion with marking their Pod privileged, I want to provide my users a means of doing that.
+
+2. As a user who needs to have their Pod access a finite set of content on a host/node in read only fashion with marking
+their Pod privileged, I want a means for doing that.
+
+3. See [this proposal](https://github.com/openshift/enhancements/pull/214) for additional user scenarios that could 
+utilize this proposal.
+
+### Implementation Details/Notes/Constraints [optional]
+
+
+#### Brief overview of CSI and the main K8S hook points
+
+ - a [CSI plugin](https://github.com/container-storage-interface/spec) is a gRPC endpoint that implements CSI services and
+ allows storage vendors to develop a storage solution that works across a number of container orchestration systems.
+ - though for our purposes, we will not provide a generic CSI plugin, but rather a special case one that is only intended
+ to work on OpenShift
+ - also for our purposes, `plugin`, `driver`, and `endpoint` seem to be the same thing based on which document you are
+ reading.
+ - k8s provides Kubernetes CSI Sidecar Containers: a set of standard containers that aim to simplify the development and 
+ deployment of CSI Drivers on Kubernetes.
+ - There are CSI volume and persistent volume types in k8s.
+ - per [k8s volume docs](https://kubernetes.io/docs/concepts/storage/volumes/#csi) the CSI types do not support direct
+ reference from a Pod and may only be referenced in a Pod via a `PersistentVolumeClaim`
```

Comment:
This is not true. That is where inline CSI volumes come - https://kubernetes-csi.github.io/docs/ephemeral-local-volumes.html

Reply:
out of date info at https://kubernetes.io/docs/concepts/storage/volumes/#csi  ... not *necessarily* surprising :-)

thanks for the clarification @gnufied 

I had noticed the inline feature touched upon in a few places, but had not dived into yet.

Aside from updating these notes to reflect the discrepancy in https://kubernetes.io/docs/concepts/storage/volumes/#csi and what inline provides, I'll read up on inline CSI and see if it has any bearing on our scenario.

Of course chime in as you see fit if you already see a place where it might.

thanks again